go/vt/vttablet/tabletmanager/restore.go

```diff
@@ -81,13 +81,23 @@ func (agent *ActionAgent) restoreDataLocked(ctx context.Context, logger logutil.
 	// Record local metadata values based on the original type.
 	localMetadata := agent.getLocalMetadataValues(originalType)
 	tablet := agent.Tablet()
-	dir := fmt.Sprintf("%v/%v", tablet.Keyspace, tablet.Shard)
 
+	keyspaceDir := tablet.Keyspace
+	keyspaceName := tablet.Keyspace
+	keyspaceInfo, err := agent.TopoServer.GetKeyspace(ctx, keyspaceName)
+	if err != nil {
+		return err
+	}
+	keyspace := keyspaceInfo.Keyspace
```

Comment:
In general, you don't need to unwrap `topo.XInfo` objects. The underlying proto struct is embedded anonymously, so you can directly access fields like `keyspaceInfo.BaseKeyspace`.